File: perl/fizzbuzz.pl
#!/usr/bin/perl

foreach( $i = 1; $i <= 100; $i++ ) {

  $three = $i % 3;
  $five  = $i % 5;

  if( $three == 0 && $five == 0 ) {
    print "FizzBuzz\n";
  }
  elsif( $three == 0 ) {
    print "Fizz\n";
  }
  elsif ( $five == 0 ) {
    print "Buzz\n";
  }
  else {
    print "$i\n";    
  }
  
}

File: java/FizzBuzz.java

public class FizzBuzz {
  
  public static void main(String []args) {
    
    for( int i = 1; i <= 100; i++ ) {
      
      int three = i % 3;
      int five  = i % 5;
      
      if( three == 0 && five == 0 ) {
        System.out.println("FizzBuzz");        
      }
      else if( three == 0 ) {
        System.out.println("Fizz");
      }
      else if ( five == 0 ) {
        System.out.println("Buzz");
      }
      else {
        System.out.println( i );
      }
      
    } // for
    
  } // main
  
} // FizzBuzz